R/smoothDegreeDays.R

```diff
@@ -36,130 +39,104 @@ smoothDegreeDays <- function(data,
+  #    Keep true historical observations unchanged
+  dataSmoothedModel <- dataFilled %>%
+    group_by(across(all_of(c("region", "variable", "tlim", "model", "rcp", "ssp")))) %>%
+    arrange(.data$period) %>%
     mutate(value = ifelse(.data[["period"]] <= endOfHistory,
                           .data[["value"]],
                           lowpass(.data[["value"]], i = nSmoothIter))) %>%
-    ungroup() %>%
-
-    # take mean over models
-    group_by(across(-all_of(c("model", "value")))) %>%
-    reframe(value = mean(.data[["value"]])) %>%
     ungroup()
 
+  # 3) Calculate ensemble mean across climate models
+  dataEnsemble <- dataSmoothedModel %>%
+    group_by(across(all_of(c("region", "variable", "tlim", "rcp", "ssp", "period")))) %>%
+    summarise(value = mean(.data[["value"]], na.rm = TRUE), .groups = "drop")
+
+  # 4) Smooth filled historical timeline
+  #    Preserve original values for true historical period (before fill-up)
+  dataEnsemble <- dataEnsemble %>%
+    group_by(across(all_of(c("region", "variable", "tlim", "rcp", "ssp")))) %>%
+    arrange(.data$period) %>%
+    mutate(valueSmooth = ifelse(.data[["period"]] <= endOfHistory,
+                                lowpass(.data[["value"]], i = nSmoothIter),
+                                .data[["value"]]),
+           value = ifelse(.data[["period"]] <= maxHistPeriod,
+                          .data[["value"]],
+                          .data[["valueSmooth"]])) %>%
+    ungroup() %>%
+    select(-"valueSmooth")
+
+  # 5) Align all scenarios (including noCC) with historical endpoint via delta shift
+  #    Calculate historical value at endOfHistory
+  histAtEnd <- dataEnsemble %>%
+    filter(.data$period == endOfHistory, .data$rcp == "historical") %>%
+    select(all_of(c("region", "variable", "tlim", "value"))) %>%
+    rename(histValue = "value")
+
+  # Calculate delta for each scenario (including noCC) at endOfHistory
+  alignDelta <- dataEnsemble %>%
+    filter(.data$period == endOfHistory) %>%
+    left_join(histAtEnd, by = c("region", "variable", "tlim")) %>%
+    filter(.data$rcp != "historical") %>%
+    mutate(delta = .data$histValue - .data$value) %>%
+    select(all_of(c("region", "variable", "tlim", "rcp", "ssp", "delta")))
+
+  # Apply delta shift to align all scenarios with historical endpoint
+  dataEnsemble <- dataEnsemble %>%
+    left_join(alignDelta, by = c("region", "variable", "tlim", "rcp", "ssp")) %>%
+    mutate(value = ifelse(!is.na(.data$delta),
+                          .data$value + .data$delta,
+                          .data$value)) %>%
+    select(-"delta")
+
+  # 6) Filter data by period: keep historical <= endOfHistory, scenarios > endOfHistory
+  dataSmooth <- rbind(
+    dataEnsemble %>%
+      filter(.data[["rcp"]] == "historical",
+             .data[["period"]] <= endOfHistory),
+    dataEnsemble %>%
+      filter(.data[["rcp"]] != "historical",
+             .data[["period"]] > endOfHistory)
+  )
```

Comment:
haven't tested it...
```suggestion
  dataSmooth <- dataEnsemble %>%
    filter((.data$rcp == "historical" & .data$period <= endOfHistory)
             | (.data$rcp != "historical" & .data$period > endOfHistory))
```
even shorter (but potentially less readable):
```suggestion
  dataSmooth <- dataEnsemble %>%
    filter(xor(.data$rcp == "historical" & .data$period > endOfHistory))
```